README.md
<img src='res/img/logo.png' width='500'></img>


# News
(latest 10 entries):
<table><tr><td width=800>

2023-07-13 'Endless-Endless-Sky' updated by 1010Todd | category: [overhauls](https://github.com/zuckung/test3/blob/main/README.md#overhauls)
<br>2023-07-13 'Amphibious-Ships' updated by Zoura | category: [ships](https://github.com/zuckung/test3/blob/main/README.md#ships)
<br>2023-07-13 'Lost in Midnight' updated by MidnightPlugins | category: [story](https://github.com/zuckung/test3/blob/main/README.md#story)
<br>2023-07-11 'Enigma Corp High DPI' updated by Amazinite | category: [story](https://github.com/zuckung/test3/blob/main/README.md#story)
<br>2023-07-11 'Kor-Fighter-Additions' updated by EjoThims | category: [ships](https://github.com/zuckung/test3/blob/main/README.md#ships)
<br>2023-07-11 'AES Omnis' updated by Darcy Manoel | category: [cheats](https://github.com/zuckung/test3/blob/main/README.md#cheats)
<br>2023-07-11 'AES Irm' updated by Darcy Manoel | category: [races](https://github.com/zuckung/test3/blob/main/README.md#races)
<br>2023-07-11 'Mata' updated by Karirawri | category: [races](https://github.com/zuckung/test3/blob/main/README.md#races)
<br>2023-07-11 'Amphibious-Ships' updated by Zoura | category: [ships](https://github.com/zuckung/test3/blob/main/README.md#ships)
<br>2023-07-11 'Midnight Expansion' updated by MidnightPlugins | category: [story](https://github.com/zuckung/test3/blob/main/README.md#story)
<br>

</td></tr></table>
Provide a directlink to your zipped plugin and we do a daily check if it is newer than the one we host here.<br>

[howto](https://github.com/zuckung/test3/blob/main/README.md#automatic-updating) | [view all news](https://github.com/zuckung/test3/blob/main/res/news.txt)

---

# Table of Contents

[Introduction](https://github.com/zuckung/test3/blob/main/README.md#introduction) 
 
[Download Information](https://github.com/zuckung/test3/blob/main/README.md#download-information)  
- [Downloading Alternative](https://github.com/zuckung/test3/blob/main/README.md#downloading-alternative)  
- [Notes](https://github.com/zuckung/test3/blob/main/README.md#notes)  
- [Known Plugin Issues](https://github.com/zuckung/test3/blob/main/README.md#known-plugin-issues)  

[Contribution](https://github.com/zuckung/test3/blob/main/README.md#contribution)  
- [Regarding Plugin Informations](https://github.com/zuckung/test3/blob/main/README.md#regarding-plugin-informations)  
- [Add your Plugin](https://github.com/zuckung/test3/blob/main/README.md#add-your-plugin)  
- [Automatic Updating](https://github.com/zuckung/test3/blob/main/README.md#automatic-updating)
- [Plugin Naming Convention](https://github.com/zuckung/test3/blob/main/README.md#plugin-naming-convention)

[Plugin Download](https://github.com/zuckung/test3/blob/main/README.md#plugin-download)  
- [Cheats](https://github.com/zuckung/test3/blob/main/README.md#cheats)  
- [Gameplay](https://github.com/zuckung/test3/blob/main/README.md#gameplay)  
- [Graphics](https://github.com/zuckung/test3/blob/main/README.md#graphics)  
- [Outfits](https://github.com/zuckung/test3/blob/main/README.md#outfits)  
- [Overhauls](https://github.com/zuckung/test3/blob/main/README.md#overhauls)  
- [Overwrites](https://github.com/zuckung/test3/blob/main/README.md#overwrites)  
- [Patches](https://github.com/zuckung/test3/blob/main/README.md#patches)  
- [Races](https://github.com/zuckung/test3/blob/main/README.md#races)  
- [Ships](https://github.com/zuckung/test3/blob/main/README.md#ships)  
- [Story](https://github.com/zuckung/test3/blob/main/README.md#story)   
- [Weapons](https://github.com/zuckung/test3/blob/main/README.md#weapons)  
- [Uncategorized](https://github.com/zuckung/test3/blob/main/README.md#uncategorized)  

---

# Introduction

A comprehensive library of 475 ancient and new plugins for the open-source game Endless Sky.

---

# Download Information

All plugins can get downloaded directly as zip-files from this page.

## Downloading Alternative

Third party tool such as [https://download-directory.github.io](https://download-directory.github.io/) can download a single folder/plugin. Paste the url to the folder url into that tool and you will be given a .zip for that folder.

For example, for the "50 cal" plugin, enter: [https://github.com/Hecter94/EndlessSky-PluginArchive/tree/main/Working/50%20cal](https://github.com/Hecter94/EndlessSky-PluginArchive/tree/main/Working/50%20cal)

Github by default only allows the entire library to be downloaded as a single .zip. Github may have issues with downloading very large file sizes, a possible workaround is using Github Desktop or Git Bash to clone the repository.

## Notes

- "Working" directory – plugins updated to work with the continous build of Endless Sky
- "Originals" directory – archived copies of original plugins
- "Nonfunctional" directory – plugins that will not function without updates

## Known Plugin Issues

<details>

### A Galaxy Far Far Away, Businessman Mod, Endless Depth, Normandy

Title screen missing/corrupted. override the title screen, using one or more of these can cause issues with it looking corrupted or "missing". Recommend to use one of these at a time, OR delete title.png from \images to restore the vanilla title screen

### Civil War, Edge of Endless, Jump to Lightspeed and Star Wars

all edit the vanilla interface, this can also cause issues and can only be fixed by editing/removing the interfaces.txt file in the plugin.

### A Wonderful Worldship

Random sprites replaced by worldships Culprit: Intended Behavior of "A wonderful worldship mod"

### Bare Ships

clears all ships from shipyards, meaning only bare versions of ships will be available, any plugins adding other ships not to custom shipyards will likely be incompatible

### Elite Sky

creates an Interceptor "bay type" which seems to make any interceptor escort jump around randomly.

### Final Frontier

sets every government swizzle to 0, also has an image named raider which conflicts with vanilla korath raider.

### Star Wars

includes a copy of vanilla data files and will likely conflict with any missions added/changed since it was created. Specifically known to conflict with changes to the Remnant storyline.

Star Wars has an image named fury which conflicts with vanilla fury.

### Recovery Ships

NPC Ships unable to assist with fuel or repairs Culprit: Likely an old version of Recovery Ship, updated 2/8/2022. Please let me know if you still see the issue.

Escorts from missions not spawning correctly Culprit: use the “safe” variant of this plugin to fix this behavior, “safe” version ships can only carry "Fighter" from vanilla and "Heavy Fighter" and "Gunship"

### The Station of Dr. Rousseau

"Bounty" Arfecta is spawned directly outside starting location. Culprit: , Mission: "Puzzle 3". No location is defined so Arfecta is spawned on first "landing".

"Hunt Dread Pirate Roberts" mission offers everywhere due to a lack of a source filter

### Edge of Endless

Overrides Luxury Accommodations and give it much higher space, bunks, and crew requirement, might make some ships that comes with it appears to have negative space and/or bunks.

Ship "Colossus" conflicts with a ship of the same name from Adamas

### Adamas

Ship "Colossus" conflicts with a ship of the same name from Edge of Endless

### ES-Restock

Gatling gun texture has empty space at 200px tall, resultuing in the outfit display being 10px lower than others

### Akasha Chronicles

Ship "Seraphim" conflicts with a ship of the same name from AES Irm

### AES Irm*

Ship "Seraphim" conflicts with a ship of the same name from Akasha Chronicles

### Mata

Ship "Spectre" conflicts with a ship of the same name from Alterra

### Alterra

Ship "Spectre" conflicts with a ship of the same name from Mata

Ship "Gladius" conflicts with a ship of the same name from Eternals

### Eternals

Ship "Gladius" conflicts with a ship of the same name from Alterra

### Cobbled Ships

Ship "Mammoth" conflicts with a ship of the same name in vanilla version 0.9.15 and later.

</details>


[back to top](https://github.com/zuckung/test3/blob/main/README.md#known-plugin-issues)

---

# Contribution

## Regarding Plugin Informations

As you may have noticed some plugins miss infos. That ranges from missing author name to full description, or just if it is still playable on the current Endless Sky version. 

Please help updating these, by either posting in discussions area or by doing a Pull Request changing the files in `/res/pluginlist/`. 
Thanks.

## Add your plugin

You want your plugin here? Fork this repository, upload your plugin and make a Pull Request. Beside that you can post it in the discussion area.

For Pull Requests, the folder structure has changed with this update. There are no more category folders for you to upload to. Instead, you create a file with the plugin information in `/res/pluginlist/` and your plugin at `Working/`, so this README.md can get auto-generated by a script. Check the other files there to see the formatting. 

If you provide the `directlink=` in the plugin-file at `res/pluginlist/`, then there is no need to upload you plugin to the `Working` directory, your plugin will be loaded on the next daily check for updates. So instead of uploading a folder with a git client, you can upload a single text-file with the web Interface.

## Automatic Updating

The files in `res/pluginlist/` which are used to generate this README.md, have a line `directlink=`, where you can put a URL to your zipped plugin. The line should look like: `directlink=https://github.com/userxyz/esplugins/releases/download/Latest/myplugin.zip` or any other directly accessible webspace.

This repository does an automated check for your directlink once a day(03:30 am utc) and if your plugin there is newer than the one hosted here, it updates it and puts it in the news box. Please read naming convention paragraph to prevent errors.

If you have a single plugin GitHub repo (with the data folder in root), you can use the auto-generated zip file (green button) as a direct update link. You can fork the official plugin template repository at [https://github.com/endless-sky/plugin-template](https://github.com/endless-sky/plugin-template).

We hope to create a centralized overview for active plugin developing with this feature.

## Plugin Naming Convention

Allthough a lot of checking and errorhandling got implemented in the scripts, there might be some bugs. To be safe, try to avoid special characters in your plugin name. 

We now have different paths and files which get addressed to the same plugin. The plugin folder is `Working/YourPlugin/`, and the plugin information file is at `res/pluginlist/YourPlugin.txt`.<br>
To keep these files related together, you have to give them the same name.


---

# Plugin Download

All Plugins (475)

[Cheats](https://github.com/zuckung/test3/blob/main/README.md#cheats) (35) | [Gameplay](https://github.com/zuckung/test3/blob/main/README.md#gameplay) (33) | [Graphics](https://github.com/zuckung/test3/blob/main/README.md#graphics) (17) | [Outfits](https://github.com/zuckung/test3/blob/main/README.md#outfits) (37)<br>
[Overhauls](https://github.com/zuckung/test3/blob/main/README.md#overhauls) (41) | [Overwrites](https://github.com/zuckung/test3/blob/main/README.md#overwrites) (1) | [Patches](https://github.com/zuckung/test3/blob/main/README.md#patches) (3) | [Races](https://github.com/zuckung/test3/blob/main/README.md#races) (64)<br>
[Ships](https://github.com/zuckung/test3/blob/main/README.md#ships) (122) | [Story](https://github.com/zuckung/test3/blob/main/README.md#story) (65) | [Weapons](https://github.com/zuckung/test3/blob/main/README.md#weapons) (39) | [Uncategorized](https://github.com/zuckung/test3/blob/main/README.md#uncategorized) (18)<br>



---

## Cheats

<p>35 plugins in this category.<p>

<details>

 

---

### AES Omnis


[AES.Omnis.zip](https://github.com/zuckung/test3/releases/download/Latest/AES.Omnis.zip) | 5.28 mb | 2023-07-11 | [view files](https://github.com/zuckung/test3/tree/main/Working/AES%20Omnis/)  
Author: Darcy Manoel | [https://github.com/Adde-Endless-Sky-Mods/-Omnis](https://github.com/Adde-Endless-Sky-Mods/-Omnis)  
Category: Cheats  

>The most extensive all-content plugin to date. It includes many opportunities to test balance or ship builds against AI driven ships or fleets, all vanilla content and the content of many plugins available to use to put together different ship builds. Many more tools are included into Omnis too!
>


Status: N/A  
Daily update check: <img src='res/img/check.png' width='15' ></img>


---

### All Content Plugin


[All.Content.Plugin.zip](https://github.com/zuckung/test3/releases/download/Latest/All.Content.Plugin.zip) | 0.35 kb | 2023-05-15 | [view files](https://github.com/zuckung/test3/tree/main/Working/All%20Content%20Plugin/)  
Author: Michael Zahniser | [https://github.com/endless-sky/all-content-plugin](https://github.com/endless-sky/all-content-plugin)  
Category: Cheats  

>This plugin was made to allow content creators to have access to all outfits in a single place, for the purpose of designing and testing outfit loadouts for new ships. Of course, it can also be used by people who want to ruin the game for themselves by getting a massive fleet without needing to work for it.
>


Status: N/A  
Daily update check: <img src='res/img/check.png' width='15' ></img>


---

### Arena


[Arena.zip](https://github.com/zuckung/test3/releases/download/Latest/Arena.zip) | 0.3 kb | 2023-05-13 | [view files](https://github.com/zuckung/test3/tree/main/Working/Arena/)  
Author: 10010101001 | [https://github.com/10010101001/Arena](https://github.com/10010101001/Arena)  
Category: Cheats  

>This plugin adds two arenas with diffrent combat styles for the player to enjoy in-game and for testing purposes (near Sagittarius A).
>


Status: N/A  
Daily update check: <img src='res/img/check.png' width='15' ></img>


---

### automata.destruction.0percent
<img src='Working/automata.destruction.0percent/icon.png' width='100'></img><br>


[automata.destruction.0percent.zip](https://github.com/zuckung/test3/releases/download/Latest/automata.destruction.0percent.zip) | 52.52 kb | 2023-06-18 | [view files](https://github.com/zuckung/test3/tree/main/Working/automata.destruction.0percent/)  
Author: zuckung | [https://github.com/zuckung/endless-sky-plugins](https://github.com/zuckung/endless-sky-plugins)  
Category: Cheats  

>(14 ship attributes changed)
>
>Modifies the self destruction chance of Sestor and Mereti ships.
>Sestor 349/109/78/71/53/40/27 and Mereti 512/256/128/64/32/16/8 ships have a self destruction value of 0.0 (0%).


Status: tested with 0.10.2  
Daily update check: <img src='res/img/check.png' width='15' ></img>


---

### automata.destruction.23percent
<img src='Working/automata.destruction.23percent/icon.png' width='100'></img><br>


[automata.destruction.23percent.zip](https://github.com/zuckung/test3/releases/download/Latest/automata.destruction.23percent.zip) | 52.54 kb | 2023-06-18 | [view files](https://github.com/zuckung/test3/tree/main/Working/automata.destruction.23percent/)  
Author: zuckung | [https://github.com/zuckung/endless-sky-plugins](https://github.com/zuckung/endless-sky-plugins)  
Category: Cheats  

>(14 ship attributes changed)
>
>Modifies the self destruction chance of Sestor and Mereti ships.
>Sestor 349/109/78/71/53/40/27 and Mereti 512/256/128/64/32/16/8 ships have a self destruction value of 0.12 (23%).


Status: tested with 0.10.2  
Daily update check: <img src='res/img/check.png' width='15' ></img>


---

### automata.destruction.51percent
<img src='Working/automata.destruction.51percent/icon.png' width='100'></img><br>


[automata.destruction.51percent.zip](https://github.com/zuckung/test3/releases/download/Latest/automata.destruction.51percent.zip) | 52.54 kb | 2023-06-18 | [view files](https://github.com/zuckung/test3/tree/main/Working/automata.destruction.51percent/)  
Author: zuckung | [https://github.com/zuckung/endless-sky-plugins](https://github.com/zuckung/endless-sky-plugins)  
Category: Cheats  

>(14 ship attributes changed)
>
>Modifies the self destruction chance of Sestor and Mereti ships.
>Sestor 349/109/78/71/53/40/27 and Mereti 512/256/128/64/32/16/8 ships have a self destruction value of 0.3 (51%).


Status: tested with 0.10.2  
Daily update check: <img src='res/img/check.png' width='15' ></img>


---

### automata.destruction.75percent
<img src='Working/automata.destruction.75percent/icon.png' width='100'></img><br>


[automata.destruction.75percent.zip](https://github.com/zuckung/test3/releases/download/Latest/automata.destruction.75percent.zip) | 52.54 kb | 2023-06-18 | [view files](https://github.com/zuckung/test3/tree/main/Working/automata.destruction.75percent/)  
Author: zuckung | [https://github.com/zuckung/endless-sky-plugins](https://github.com/zuckung/endless-sky-plugins)  
Category: Cheats  

>(14 ship attributes changed)
>
>Modifies the self destruction chance of Sestor and Mereti ships.
>Sestor 349/109/78/71/53/40/27 and Mereti 512/256/128/64/32/16/8 ships have a self destruction value of 0.5 (75%).


Status: tested with 0.10.2  
Daily update check: <img src='res/img/check.png' width='15' ></img>


---

### Bit and Bobs


[Bit.and.Bobs.zip](https://github.com/zuckung/test3/releases/download/Latest/Bit.and.Bobs.zip) | 0.33 kb | 2023-05-15 | [view files](https://github.com/zuckung/test3/tree/main/Working/Bit%20and%20Bobs/)  
Author: N/A | N/A[]()  
Category: Cheats  

>N/A
>


Status: N/A  
Daily update check: <img src='res/img/cross.png' width='15' ></img>


---

### Casual Mod


[Casual.Mod.zip](https://github.com/zuckung/test3/releases/download/Latest/Casual.Mod.zip) | 0.32 kb | 2023-05-15 | [view files](https://github.com/zuckung/test3/tree/main/Working/Casual%20Mod/)  
Author: PPhoto  | [https://github.com/pphotography/EndlessSky_CasualMod](https://github.com/pphotography/EndlessSky_CasualMod)  

res/news.txt
2023-07-13 'Endless-Endless-Sky' updated by 1010Todd | category: [overhauls](https://github.com/zuckung/test3/blob/main/README.md#overhauls)
2023-07-13 'Amphibious-Ships' updated by Zoura | category: [ships](https://github.com/zuckung/test3/blob/main/README.md#ships)
2023-07-13 'Lost in Midnight' updated by MidnightPlugins | category: [story](https://github.com/zuckung/test3/blob/main/README.md#story)
2023-07-11 'Enigma Corp High DPI' updated by Amazinite | category: [story](https://github.com/zuckung/test3/blob/main/README.md#story)
2023-07-11 'Kor-Fighter-Additions' updated by EjoThims | category: [ships](https://github.com/zuckung/test3/blob/main/README.md#ships)
2023-07-11 'AES Omnis' updated by Darcy Manoel | category: [cheats](https://github.com/zuckung/test3/blob/main/README.md#cheats)
2023-07-11 'AES Irm' updated by Darcy Manoel | category: [races](https://github.com/zuckung/test3/blob/main/README.md#races)
2023-07-11 'Mata' updated by Karirawri | category: [races](https://github.com/zuckung/test3/blob/main/README.md#races)
2023-07-11 'Amphibious-Ships' updated by Zoura | category: [ships](https://github.com/zuckung/test3/blob/main/README.md#ships)
2023-07-11 'Midnight Expansion' updated by MidnightPlugins | category: [story](https://github.com/zuckung/test3/blob/main/README.md#story)
2023-07-11 'MechSummoner' updated by Zoura | category: [uncategorized](https://github.com/zuckung/test3/blob/main/README.md#uncategorized)
2023-07-11 'Lost in Midnight' updated by MidnightPlugins | category: [story](https://github.com/zuckung/test3/blob/main/README.md#story)
2023-07-11 'AES Aben' updated by Darcy Manoel | category: [races](https://github.com/zuckung/test3/blob/main/README.md#races)
2023-07-10 'Amphibious-Ships' updated by Zoura | category: [ships](https://github.com/zuckung/test3/blob/main/README.md#ships)
2023-07-09 'Lost in Midnight' updated by MidnightPlugins | category: [story](https://github.com/zuckung/test3/blob/main/README.md#story)
2023-07-08 'test file' updated by tester | category: [uncategorized](https://github.com/zuckung/test3/blob/main/README.md#uncategorized)
2023-07-08 'Midnight Scrapyard' updated by MidnightPlugins | category: [story](https://github.com/zuckung/test3/blob/main/README.md#story)
2023-07-08 'Amphibious-Ships' updated by Zoura | category: [ships](https://github.com/zuckung/test3/blob/main/README.md#ships)
2023-07-08 'Lost in Midnight' updated by MidnightPlugins | category: [story](https://github.com/zuckung/test3/blob/main/README.md#story)
2023-07-07 'Lemurias Extra ES Stuff' updated by a-random-lemurian | category: [story](https://github.com/zuckung/test3/blob/main/README.md#story)
2023-07-06 'additional.command.buttons' updated by zuckung | category: [graphics](https://github.com/zuckung/test3/blob/main/README.md#graphics)
2023-07-06 'Amphibious-Ships' updated by Zoura | category: [ships](https://github.com/zuckung/test3/blob/main/README.md#ships)
2023-07-06 'test file' updated by tester | category: [uncategorized](https://github.com/zuckung/test3/blob/main/README.md#uncategorized)
2023-07-06 'Adamas Project' updated by todd1010 | category: [races](https://github.com/zuckung/test3/blob/main/README.md#races)
2023-07-05 'test file' updated by N/A | category: [uncategorized](https://github.com/zuckung/test3/blob/main/README.md#uncategorized)
2023-07-05 'Amphibious-Ships' updated | category: [ships](https://github.com/zuckung/test3/blob/main/README.md#ships)
2023-07-04 'test file' updated | category: [uncategorized](https://github.com/zuckung/test3/blob/main/README.md#uncategorized)
2023-07-04 'World Forge' updated | category: [cheats](https://github.com/zuckung/test3/blob/main/README.md#cheats)
2023-07-04 'Galactic War' updated | category: [overhauls](https://github.com/zuckung/test3/blob/main/README.md#overhauls)
2023-07-04 'Amphibious-Ships' updated | category: [ships](https://github.com/zuckung/test3/blob/main/README.md#ships)
2023-07-04 'Bulkheads-and-Battleships' updated | category: [uncategorized](https://github.com/zuckung/test3/blob/main/README.md#uncategorized)
2023-07-04 'StarProtector-Endless-Sky' updated | category: [ships](https://github.com/zuckung/test3/blob/main/README.md#ships)
2023-07-04 'Midnight Scrapyard' updated | category: [story](https://github.com/zuckung/test3/blob/main/README.md#story)
2023-07-04 'The Evora Navaiya' updated | category: [races](https://github.com/zuckung/test3/blob/main/README.md#races)
2023-07-04 'Sestor-Expansion' updated | category: [ships](https://github.com/zuckung/test3/blob/main/README.md#ships)
2023-07-04 'Lost in Midnight' updated | category: [story](https://github.com/zuckung/test3/blob/main/README.md#story)
2023-07-03 'Derogam' updated | category: [ships](https://github.com/zuckung/test3/blob/main/README.md#ships)
2023-07-03 'Beyond the Sky' updated | category: [overhauls](https://github.com/zuckung/test3/blob/main/README.md#overhauls)
2023-07-03 'Blended Ships' updated | category: [ships](https://github.com/zuckung/test3/blob/main/README.md#ships)
2023-07-02 'Mega Freight' updated | category: [ships](https://github.com/zuckung/test3/blob/main/README.md#ships)
2023-06-21 'galactic.capital.investment' updated | category: [gameplay](https://github.com/zuckung/test3/blob/main/README.md#gameplay)
2023-06-21 'too.many.asteroids' updatedcategory=overhauls
2023-06-21 'additional.command.buttons' updatedcategory=graphics
2023-06-21 'snowfeather.robotics' updatedcategory=outfits
2023-06-13 'unique.fix' updated
2023-06-12 repo update: directlink and news


res/src/makemd.py
import os
import requests
from datetime import datetime
import PIL
from PIL import Image

iconpng = assetfile = pluginname =  lastmodified = size = author = website = category = status = description = pluginnameurl = news = updatecheck =""
allplugins = cheats = gameplay = graphics = outfits = overhauls = overwrites = patches = races = ships = story = weapons = uncategorized = 0
categories = ["Cheats", "Gameplay", "Graphics", "Outfits", "Overhauls", "Overwrites", "Patches", "Races", "Ships", "Story", "Weapons", "Uncategorized"]
plist = []

# check for local testing
if os.getcwd() == "/storage/emulated/0/Download/mgit/test3/res/src":
	os.chdir("../../")

# read paths and files
with open("res/config.txt") as f:
	for line in f:
		line = str((line.strip()))
		if line.find("indexfile") == 0:	# i.e. indexfile = README.md | can get changed to whatever file you want the script output go
			indexfile = line.split(" = ")[1]
		if line.find("template") == 0:	# i.e. template = res/template.txt | a template how the indexfile should look, can be edited
			template = line.split(" = ")[1]
		if line.find("listfolder") == 0:	# i.e. listfolder = res/pluginlist/ | folder to where the plugin descriptions files are
			listfolder = line.split(" = ")[1]
		if line.find("pathtoplugins") == 0:	# i.e. pathtoplugins = myplugins/ | folder where the plugin folder and files are
			pathtoplugins = line.split(" = ")[1]
		if line.find("webroot") == 0:	# i.e. webroot = https://github.com/zuckung/test3/blob/main/
			webroot = line.split(" = ")[1]	
		if line.find("pluginurl") == 0:	# i.e. pluginurl = https://github.com/zuckung/test3/tree/main/Working/All%20Plugins/
			pluginurl = line.split(" = ")[1]
		if line.find("assetfullpath") == 0:	# i.e. assetfullpath = https://github.com/zuckung/test3/releases/download/Latest/
			assetfullpath = line.split(" = ")[1]
			
def replacevar(string): # replaces %variables% in header template with real values
	if cat == "AllPlugins":
		string = string.replace("%amount%", str(allplugins))
	elif cat == "Cheats":
		string = string.replace("%amount%", str(cheats))
	elif cat == "Gameplay":
		string = string.replace("%amount%", str(gameplay))
	elif cat == "Graphics":
		string = string.replace("%amount%", str(graphics))
	elif cat == "Outfits":
		string = string.replace("%amount%", str(outfits))
	elif cat == "Overhauls":
		string = string.replace("%amount%", str(overhauls))
	elif cat == "Overwrites":
		string = string.replace("%amount%", str(overwrites))
	elif cat == "Patches":
		string = string.replace("%amount%", str(patches))
	elif cat == "Races":
		string = string.replace("%amount%", str(races))
	elif cat == "Ships":
		string = string.replace("%amount%", str(ships))
	elif cat == "Story":
		string = string.replace("%amount%", str(story))
	elif cat == "Weapons":
		string = string.replace("%amount%", str(weapons))
	elif cat == "Uncategorized":
		string = string.replace("%amount%", str(uncategorized))
	string = string.replace("%allplugins%", str(allplugins))
	string = string.replace("%cheats%", str(cheats))
	string = string.replace("%gameplay%", str(gameplay))
	string = string.replace("%graphics%", str(graphics))
	string = string.replace("%outfits%", str(outfits))
	string = string.replace("%overhauls%", str(overhauls))
	string = string.replace("%overwrites%", str(overwrites))
	string = string.replace("%patches%", str(patches))
	string = string.replace("%races%", str(races))
	string = string.replace("%ships%", str(ships))
	string = string.replace("%story%", str(story))
	string = string.replace("%weapons%", str(weapons))
	string = string.replace("%uncategorized%", str(uncategorized))
	return string

def replacevarp(string):
	string = string.replace("%assetfullpath%", assetfullpath)
	string = string.replace("%assetfile%", assetfile)
	string = string.replace("%pluginname%", pluginname)
	string = string.replace("%assetfile%", assetfile)
	string = string.replace("%lastmodified%", lastmodified)
	string = string.replace("%size%", size)
	string = string.replace("%pluginurl%", pluginurl)
	string = string.replace("%author%", author)
	string = string.replace("%webroot%", webroot)
	string = string.replace("%indexfile%", indexfile)
	string = string.replace("%news%", news)
	string = string.replace("%updatecheck%", updatecheck)
	
	
	if website == "N/A": # for prevent [N/A](N/A) links
		websitecheck = "N/A"
		websitelink = ""
	else:
		websitecheck = ""
		websitelink = website
	string = string.replace("%website%", websitelink)
	string = string.replace("%websitecheck%", websitecheck)
	string = string.replace("%category%", category)
	string = string.replace("%status%", status)
	string = string.replace("%iconpng%", str(iconpng))
	string = string.replace("%pluginnameurl%", pluginnameurl)	
	string = string.replace("%description%", description)
	return string
				
# counting categories of files in  pluginlistfolder
entries = os.listdir(pathtoplugins)
entries.sort(key=str.lower)
for entry in entries:
	allplugins += 1
	# if pluginlist file exists, read it
	if os.path.exists(listfolder + entry + ".txt") == True:
		with open(listfolder + entry + ".txt", "r") as file1:
			cat = file1.readline()
			cat = file1.readline()
			cat = file1.readline()
			cat = file1.readline().split("=")[1].replace("\n", "")
			if cat == "cheats":
				cheats += 1
			elif cat == "gameplay":
				gameplay += 1
			elif cat == "graphics":
				graphics += 1
			elif cat == "outfits":
				outfits += 1
			elif cat == "overhauls":
				overhauls += 1
			elif cat == "overwrites":
				overwrites += 1
			elif cat == "patches":
				patches += 1
			elif cat == "races":
				races += 1
			elif cat == "ships":
				ships += 1
			elif cat == "story":
				story += 1
			elif cat == "weapons":
				weapons += 1
			else:
				uncategorized += 1
	else:
		uncategorized += 1
		cat = "uncategorized"
		# if no pluginlist file exists, create an empty one	
		with open(listfolder + entry + ".txt" , "w") as file1:
			file1.writelines("author=N/A\nwebsite=N/A\ndirectlink=N/A\ncategory=N/A\nstatus=N/A\ndescription=N/A\n")
		print(entry + ".txt CREATED! because plugin was there, but no listfile!")
	with open(listfolder + entry + ".txt", "r") as file1:
		text = file1.read()
		plist.append(str(cat) + "\n" + entry + "\n" + text) # create a list of all listfiles

# reading template file and splitting it to the 3 templates
with open(template, "r") as file1:
	temp = file1.read()
pos = temp.find("%%% template for category below this point %%%")
temphead = temp[:pos] # set header string
foot = temp[pos + 47:]	
pos = foot.find("%%% template for plugin entry below this point %%%")
temptempcat = foot[:pos] # set category string
tempplug = foot[pos + 50:] # set plugin string
pos = temptempcat.find("%pluginhere%")
tempcatup = temptempcat[:pos] # upper half of template category
tempcatdown = temptempcat[pos +12:] # lower half of template string


with open("res/news.txt", "r") as file1: # reading and formating lines
	newslist = file1.readlines()
for i in range(10): # define amount of news to 10
	if i <= len(newslist)-1:
		news = news + newslist[i] + "<br>"

# writing the md file
with open("res/errorlog.txt", "w") as filerr: # resetting errorlog.txt
	filerr.writelines("header request error log:\n")
with open(indexfile, "w") as file1:
	temphead = replacevar(temphead)
	temphead = replacevarp(temphead)
	file1.writelines(temphead) # writer header template
	for cat in categories: # for each category
		tempcatupt = tempcatup.replace("%category%", cat)
		tempcatupt = replacevar(tempcatupt)
		tempcatupt = replacevarp(tempcatupt)
		tempcatdownt = tempcatdown.replace("%category%", cat)
		tempcatdownt = replacevarp(tempcatdownt)
		tempcatdownt = replacevar(tempcatdownt)
		file1.writelines(tempcatupt) # write upper category template
		for p in plist: # listfiles list element, created in counting block... 'array' of all plugins
			pos = p.find("\n")
			catcomp = p[:pos]
			catcomp = catcomp.strip() # get category for comparing with actual category
			if catcomp == "N/A":
				catcomp = "Uncategorized"
			if catcomp.capitalize() == cat:
				# get variables out of the list item p
				description = p.split("\n")
				if description[0] == "N/A":
					category = description[0]
				else:
					category = description[0].capitalize() 
				pluginname = description[1]
				pluginnameurl = pluginname.replace(" ", "%20")
				author = description[2][7:] 
				website = description[3][8:] 
				directlink = description[4][11:]
				status = description[6][7:]
				for x in range(0, 7):
					description.pop(0)
				description[0] = description[0].replace("description=", "")
				alllines = ""
				for lines in description:
					alllines = alllines + ">" + lines + "\n"
				description = alllines
				
				if os.path.exists(pathtoplugins + pluginname + "/icon.png") == True: # check for icon.png, resize it, or hide it
					im = Image.open(pathtoplugins + pluginname + "/icon.png")
					w, h = im.size
					iconpath = pathtoplugins + pluginname
					iconpath = iconpath.replace("&", "%26")
					iconpath = iconpath.replace("'", "%27")
					iconpath = iconpath.replace("(", "%28")
					iconpath = iconpath.replace(")", "%29")
					iconpath = iconpath.replace(",", "%2C")
					if h > w:
						iconpng = "<img src='"+ iconpath + "/icon.png' height='100'></img><br>\n"
					else:
						iconpng = "<img src='"+ iconpath + "/icon.png' width='100'></img><br>\n"
				else:
					iconpng = ""
				
				# get last modified date from the assetfiles
				withdots = pluginname.replace(" ", ".") 
				withdots = withdots.replace("'", ".")
				withdots = withdots.replace(",", ".") 
				withdots = withdots.replace("(", ".") 
				withdots = withdots.replace(")", ".") 
				withdots = withdots.replace("&", ".") 
				withdots = withdots.replace("...", ".")
				withdots = withdots.replace("..", ".")
				if withdots[len(withdots)-1] == ".":
					withdots = withdots[:len(withdots)-1]
				try:
					response = requests.head(assetfullpath + withdots + ".zip", allow_redirects=True)
					response.raise_for_status()
				except requests.exceptions.HTTPError as err:
					print(err)
					lastmodified = "N/A"
					size = "N/A" 
					filerr = open("res/errorlog.txt", "a")
					filerr.writelines(str(err) + " \n")
					filerr.close
				else:
					#response = requests.head(assetfullpath + withdots + ".zip", allow_redirects=True)
					#response.raise_for_status()
					modif = response.headers['Last-Modified']
					datetime_object = datetime.strptime(modif, '%a, %d %b %Y %H:%M:%S %Z')
					lastmodified = str(datetime_object.date())
					# get file size of the assetfiles in kb or mb
					assetsize = int(response.headers['Content-Length']) / 1024
					form = " kb"
					if assetsize > 1024:
						assetsize = assetsize / 1024
						form = " mb"
					size = str(round(assetsize, 2)) + form
					if directlink != "N/A": # check if plugin has direct updating and color a dot
						updatecheck = "<img src='res/img/check.png' width='50' ></img>"
					else:
						updatecheck = "<img src='res/img/cross.png' width='50' ></img>"
				assetfile =  withdots + ".zip"
				file1.writelines(replacevarp(tempplug)) # write plugin template entry, exchanging %variables%
		file1.writelines(tempcatdownt) # write lower category template
